packages/dds/tree/src/shared-tree/sharedTreeChangeCodecs.ts

```diff
@@ -58,15 +59,30 @@ export function makeSharedTreeChangeCodec(
 	return {
 		encode: (change, context) => {
 			const changes: EncodedSharedTreeInnerChange[] = [];
+			let updatedSchema: TreeStoredSchema | undefined;
```

Comment:
You could make this typed as Mutable<SchemaAndPolicy> (and never undefined) and initialize it to

`{
	policy:
		context.schema !== undefined
			? context.schema.policy
			: defaultSchemaPolicy,
	schema: context.schema,
}`

and then mutate the schema field in schema branch.

Reply:
I couldn't quite get this to work as there was no `defaultSchema` we could use (similarly to `defaultSchemaPolicy` used for policy) if `context.schema` is undefined.

If we do the following, there is no good default value for the schema we can initialize to when `context.schema===undefined`, since schema is required in the type `SchemaAndPolicy`

`{ policy: context.schema !== undefined ? context.schema.policy : defaultSchemaPolicy, schema: context.schema.schema, }`

Reply:
ah, fair. thanks!